pkg/systemd/reporting.jsx

```diff
@@ -0,0 +1,467 @@
+        get_problem_property(this.props.problem, "CanBeReported")
+                .done((reportable) => {
+                    if (!reportable) {
+                        this.setState({ problemState: ProblemState.UNREPORTABLE });
+                    }
+                });
+    }
+
+    onCancelButtonClick(event) {
+        this.state.process.close("canceled");
+    }
+
+    onReportButtonClick(event) {
+        if (this.state.problemState === ProblemState.REPORTING) {
+            return;
+        }
+
+        const process = cockpit.spawn(["reporter-ureport", "-d", this.props.problem.ID],
+                                      {
+                                          err: "out",
+                                          superuser: "true",
+                                      })
+                .always(() => {
+                    this.updateStatusFromBus();
+                })
+                .done(() => {
+                    this.setState({ problemState: ProblemState.REPORTED, });
+                })
+                .fail((ex) => {
+                    this.setState({ problemState: ProblemState.REPORTABLE, });
+                })
+                .stream((data) => {
+                    this.setState({ message: data, });
+                });
+
+        this.setState({
+            problemState: ProblemState.REPORTING,
+            process: process,
+        });
+    }
+
+    render() {
+        return <WorkflowRow label="ABRT Analytics"
+                            message={this.state.message}
+                            onCancelButtonClick={this.onCancelButtonClick}
+                            onReportButtonClick={this.onReportButtonClick}
+                            problemState={this.state.problemState}
+                            reportLinks={this.state.reportLinks} />;
+    }
+}
+
+const reportd_client = cockpit.defer();
+const permission = cockpit.permission({ admin: true });
+permission.addEventListener("changed", () => {
+    // reportd may use ABRT API that requires authorization, but it cannot
+    // be given using polkit, as the calling process will always be reportd
+    // and not cockpit-bridge. However, UID 0 is always authorized, hence
+    // the spawning of the system service and using the system bus here.
+    //
+    // TODO: Only use system bus when reportd is merged into ABRT
+    //       (https://github.com/abrt/reportd/issues/8).
+    const bus_type = permission.allowed ? "system" : "session";
+    const client = cockpit.dbus("org.freedesktop.reportd", { bus: bus_type });
+    client.wait()
+            .done(() => {
+                reportd_client.resolve(client);
+            })
+            .fail(() => {
+                reportd_client.reject();
+            });
+});
+
+class BusWorkflowRow extends React.Component {
+    constructor(props) {
+        super(props);
+
+        let label = this.props.workflow[1];
+        const match = /(?:Report to) (.*)/.exec(this.props.workflow[1]);
```

Comment:
Out of interest, this `Report to` is name of workflow - it would never be localized on the system?

Reply:
ok, it comes from reportd's `GetWorkflows` which always passes it in English, right?

Reply:
Ah, crap, they are localizable. I don’t know why I thought otherwise. So this complicates things a little.

`Report to Fedora                     [ Report ]` - this looks a bit awkward now.

Reply:
When we use [spawn](https://cockpit-project.org/guide/latest/cockpit-spawn.html) we can set environment (LC_ALL=C) to work around this issue. Maybe something similar could be solution for this problem? Or introduce short name in workflows?
But if this is bigger problem, I guess then follow up is fine.